test/models/profile_host_test.rb

```diff
@@ -1,11 +1,27 @@
 # frozen_string_literal: true
 
 require 'test_helper'
+require 'sidekiq/testing'
 
 class ProfileHostTest < ActiveSupport::TestCase
   setup do
     @profile = profiles(:one)
     @host = hosts(:one)
     @profile_host = ProfileHost.new(profile: @profile, host: @host)
   end
+
+  test 'destroys associated hosts if host has no more policies assigned' do
+    profile = Profile.create(name: 'foo', benchmark: benchmarks(:one),
+                          ref_id: 'foo')
+    host = Host.create(profiles: [profile], name: 'bar',
+                       account: accounts(:one))
+    assert_equal 0, DeleteHost.jobs.size
```

Comment:
You should put `DeleteHost.clear` in the setup method to ensure no other tests have queued jobs on this test object.

Reply:
Updated